cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/PermissionManagerSpi.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+import io.cdap.cdap.api.security.AccessException;
+import io.cdap.cdap.proto.id.EntityId;
+import io.cdap.cdap.proto.security.Authorizable;
+import io.cdap.cdap.proto.security.GrantedPermission;
+import io.cdap.cdap.proto.security.Permission;
+import io.cdap.cdap.proto.security.Principal;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Interface to manage {@link Permission} on entities. Depending on authorization extension can
+ * provide management access or not depending on security type. In the latter case permission must
+ * be managed using native tools for the security environment.
+ */
+public interface PermissionManagerSpi {
```

Comment:
It seems that PermissionManagerSpi/RoleControllerSpi are missing principal of the user doing the action (must be coming from authenticationContext.getPrincipal()). Could you please add it? Note that even in cases when the principal parameter is already there (e.g. in `grant`) it's not the caller's principal, but the principal we grant to, so we need additional parameter. Overall all calls need `Principal caller` parameter that is passed from `authenticationContext.getPrincipal()` when we convert API to SPI.

Reply:
The `AccessControllerInstantiator` which creates the SPI  has the `authenticationContext` 

can we utilise/expose that info directly ?

Or how do we get the caller info in some APIs like `DelegatingRoleController` ?

Reply:
DelegatingRoleController should have access to AuthenticationContext, is not it?